platform/barefoot/sonic-platform-modules-ufispace/s9180-32x/sonic_platform/chassis.py

```diff
@@ -0,0 +1,238 @@
+        start_time = time.time()
+        forever = False
+
+        if timeout == 0:
+            forever = True
+        elif timeout > 0:
+            timeout = timeout / float(1000)  # Convert to secs
+        else:
+            print("get_change_event:Invalid timeout value", timeout)
+            return False, change_event_dict
+
+        end_time = start_time + timeout
+        if start_time > end_time:
+            print(
+                "get_change_event:" "time wrap / invalid timeout value",
+                timeout,
+            )
+            return False, change_event_dict  # Time wrap or possibly incorrect timeout
+        try:
+            while timeout >= 0:
+                # check for sfp
+                sfp_change_dict = self.get_transceiver_change_event()
+                # check for fan
+                # fan_change_dict = self.get_fan_change_event()
+                # check for voltage
+                # voltage_change_dict = self.get_voltage_change_event()
+
+                if sfp_change_dict:
+                    change_event_dict["sfp"] = sfp_change_dict
+                    # change_event_dict["fan"] = fan_change_dict
+                    # change_event_dict["voltage"] = voltage_change_dict
+                    return True, change_event_dict
+                if forever:
+                    time.sleep(1)
+                else:
+                    timeout = end_time - time.time()
+                    if timeout >= 1:
+                        time.sleep(1)  # We poll at 1 second granularity
+                    else:
+                        if timeout > 0:
+                            time.sleep(timeout)
+                        return True, change_event_dict
+        except Exception as e:
+            print(e)
+        print("get_change_event: Should not reach here.")
+        return False, change_event_dict
+
+    def get_transceiver_change_event(self):
+        current_port_dict = {}
+        ret_dict = {}
+
+        # Check for OIR events and return ret_dict
+        for index in range(self.platform_inventory['num_ports']):
+            if self._sfp_list[index].get_presence():
+                current_port_dict[index] = self.STATUS_INSERTED
+            else:
+                current_port_dict[index] = self.STATUS_REMOVED
+
+        if len(self.port_dict) == 0:       # first time
+            self.port_dict = current_port_dict
+            return {}
+
+        if current_port_dict == self.port_dict:
+            return {}
+
+        # Update reg value
+        for index, status in current_port_dict.items():
+            if self.port_dict[index] != status:
+                ret_dict[index] = status
+                #ret_dict[str(index)] = status
+        self.port_dict = current_port_dict
+        for index, status in ret_dict.items():
+            if int(status) == 1:
+                pass
+                #self._sfp_list[int(index)].check_sfp_optoe_type()
+        return ret_dict
+
+    def get_sfp(self, index):
```

Comment:
This might not be required as he parent class chassi_base.py provide the same definition of the func.

Reply:
Ok. I will remove this function.